### DISCUSSION TITLE
Where can I edit oas but Swagger?

### DISCUSSION THREAD

**Discussion by razb-viola** [OAI/OpenAPI-Specification#3407] [Q&A] [answered]
I saw that Swagger is a tool to build my oas, but Swagger is a private company, while oas is a public standard and open source. How can I use oas without any intervention of a private company?

I need code autocompletion and the ability to produce docs.

If this is something that is not included within the oas, just say. 

Thanks.

**Comment by MikeRalphson** [comment, marked as answer]
There are many open-source tools for editing OAS descriptions listed at https://tools.openapis.org/

**Reply by razb-viola** [reply]
Thanks, got any flavored?

**Reply by MikeRalphson** [reply]
We don't endorse specific tools, but you might want to look at the number of GitHub stars/forks as some kind of popularity metric.
